### ISSUE TITLE
dartpad doesn't print some uncaught type errors

### ISSUE THREAD

**Issue by jcollins-g** [dart-lang/dart-pad#836]
Given the example code:
```dart
import"dart:async";
main() async {
  var ffi = new Future<Future<int>>.value(new Future<int>.value(42));
  print(ffi.runtimeType);
  var fi = await ffi;
  print(fi.runtimeType);
  var i = await fi;
  print(i);
}
```

When run, the program appears to complete after the first "print" without error, but there's an error on the console:

```
VM160:717 Uncaught Error: Assertion failed
    at Object.wrapException (<anonymous>:717:17)
    at Object.assertHelper (<anonymous>:1225:17)
    at _Future__asyncComplete_closure.call$0 (<anonymous>:2805:11)
    at Object._microtaskLoop (<anonymous>:2116:21)
    at StaticClosure.dart._startMicrotaskLoop (<anonymous>:2122:11)
    at _AsyncRun__initializeScheduleImmediate_internalCallback.call$1 (<anonymous>:2257:11)
    at invokeClosure (<anonymous>:851:26)
    at MutationObserver.<anonymous> (<anonymous>:867:18)
```

**Comment by jcollins-g** [comment]
@sigmundch

**Comment by jcollins-g** [comment]
@lrhn

**Comment by lrhn** [comment]
Assertion fixed in 76db06cc249898e2aebd3f9a6bcf539b5db3eb86
